teuthology/nuke.py

```diff
@@ -361,6 +367,101 @@ def synch_clocks(remotes):
         log.info('Waiting for clock to synchronize on %s...', name)
         proc.wait()
 
+#
+# A delay, in seconds, that is significantly longer than
+# any kind of OpenStack server creation / deletion / etc.
+#
+OPENSTACK_DELAY = 30 * 60
+
+def stale_openstack_instances(ctx, instances, locked_nodes):
+    for (name, instance) in instances.iteritems():
+        i = OpenStackInstance(name)
+        if (i.get_created() >
+            ctx.teuthology_config['max_job_time'] + OPENSTACK_DELAY):
+            log.info("stale-openstack: instance " + i['name'] + " destroyed" +
```

Comment:
More paste tense before destruction. Seems like these strings would benefit from templating either using the ``%s``-style replacement or ``str.format()`` - then, this message could be reused and extended below optionally